aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -409,6 +410,15 @@ void _assert_async_msg_cpu(const Tensor& self, c10::string_view assert_msg) {
   TORCH_CHECK(native::is_nonzero(self), assert_msg != "" ? assert_msg : "Assertion is failed");
 }
 
+Tensor _functional_assert_async_msg_cpu(
+  const Tensor& self,
+  c10::string_view assert_msg,
+  const Tensor& dep_token) {
+  _assert_async_msg_cpu(self, assert_msg);
+  return dep_token;
+}
```

Comment:
Mentioned this over chat to Tugsuu but just to point out the problem: PyTorch has a problem if your non-CompositeImplicitAutograd operator has incorrect alias information. The schema promised that the operator was functional (and that none of the outputs alias the inputs), but we are returning an input (the world token) directly as an output.

This can end up triggering asserts (some of which can only be see in a DEBUG build).

The fix for this is to just `return dep_token.clone()`. The dep_token should be a Tensor without storage anyways, so the clone should be cheap.